troposphere/static/js/components/modals/allocationSource/NoAllocationSourceModal.react.js

```diff
@@ -0,0 +1,310 @@
+import React from 'react';
+import Backbone from 'backbone';
+import _ from 'underscore';
+
+import BootstrapModalMixin from 'components/mixins/BootstrapModalMixin.react';
+import stores from 'stores';
+import globals from 'globals';
+import SelectMenu from 'components/common/ui/SelectMenu2.react';
+import EventActions from "actions/EventActions";
+import EventConstants from "constants/EventConstants";
+
+const DefaultModalView = React.createClass({
+    displayName: "NoAllocationSourceDefaultModalView",
+
+    propTypes: {
+        instances: React.PropTypes.instanceOf(Backbone.Collection).isRequired,
+        projects: React.PropTypes.instanceOf(Backbone.Collection).isRequired,
+        allocationSources: React.PropTypes.instanceOf(Backbone.Collection).isRequired,
+        onConfirm: React.PropTypes.func.isRequired
+    },
+
+    getInitialState() {
+        return this.getStateFromProps(this.props);
+    },
+
+    componentWillReceiveProps(props) {
+        this.setState(this.getStateFromProps(props));
+    },
+
+    getStateFromProps(props) {
+        let { allocationSources, instances } = this.props;
+
+        // This is a structure for the pairing of (instance, allocationSource)
+        // Ex. { instance.id:  { instance, allocationSource }}
+        let instanceAllocations = {};
+        let defaultAllocation = allocationSources.first();
+
+        // Create a default pairing for each instance
+        instances.forEach(instance => {
+            instanceAllocations[instance.id] = {
+                allocationSource: defaultAllocation,
+                instance
+            }
+        })
+
+        return {
+            instanceAllocations,
+        }
+    },
+
+    onConfirm() {
+        // instanceAllocations takes the form of:
+        // {
+        //     instanceId: { allocationSource, instance }
+        // }
+        let { instanceAllocations } = this.state;
+
+        // Flatten it to:
+        // [ { allocationSource, instance } ]
+        let flattened = _.values(instanceAllocations);
+
+        this.props.onConfirm(flattened);
+    },
+
+    renderInstance(instance) {
+        let { allocationSources } = this.props;
+        let { allocationSource } = this.state.instanceAllocations[instance.id];
+
+        return (
+        <div
```

Comment:
This should be a list item both semantically and because we are rendering this into a `ul` element below.